### PR TITLE
feat(sandbox): make the standalone network lifecycle a fail-closed required service

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## What changed

Makes the standalone sandbox network lifecycle a typed, fail-closed required service, the substrate the upcoming `pipelock guard` preview relies on.

`StandaloneLaunchConfig` gains `RequireNetNS` and `RequireProxyHandler`. A required nil handler, a required namespace when best-effort is set, or unavailable user namespaces are rejected before the child process starts, so the debug-only direct-forward path is unreachable for a caller that requires scanning.

`BridgeProxy.Serve` now returns an error and returns nil only on intentional Close or context cancellation. A lost parent socket is recorded as a fatal failure that wakes Serve, and the standalone init treats bridge death as a non-zero exit so the parent terminates the whole process group. When the agent exits first, the bridge is closed and its terminal result consumed before success is declared, so a failure already in progress is never masked by a select race.

The per-invocation control directory is created 0700, and a pre-existing group- or other-enterable directory is refused rather than silently re-permissioned.

`Preflight` is split into `PreflightWithRequirements` with a backward-compatible legacy strict wrapper, so network, handler, Landlock, and seccomp requirements are reported independently.

## Why

The guard preview requires that a sandboxed process either reaches the network through Pipelock's scanning bridge or reaches it not at all. That requires the namespace and bridge to be mandatory rather than best-effort, a nil scanning handler to be impossible, and bridge death to fail the session loudly rather than degrade silently.

## Scope

The new requirement fields are opt-in and off by default; ordinary `pipelock sandbox` behavior is unchanged. The MCP bridge path (`pipelock mcp proxy`) still discards the new Serve error and is a separate lifecycle concern, not addressed here.

## Summary by CodeRabbit

* **New Features**
  * Added configurable requirements for network isolation and proxy support during sandbox launches.
  * Preflight checks now report individual security and isolation capabilities and support best-effort, strict, and required modes.
* **Bug Fixes**
  * Sandbox startup now surfaces bridge and connection failures instead of silently continuing.
  * Improved shutdown handling prevents hangs and preserves agent exit results.
* **Security**
  * Control directories are validated as private, unpredictable, and securely permissioned.
* **Validation**
  * Added coverage for requirement checks, failure propagation, lifecycle handling, and clean shutdowns.

### CODE REVIEW

File: internal/sandbox/standalone_test.go [hunk 1]
```diff
@@ -116,6 +116,35 @@ func TestLaunchStandalone_RejectsInvalidWorkspace(t *testing.T) {
...
+		Workspace:           workspace,
+		RequireProxyHandler: true,
+	})
+	if err == nil || !strings.Contains(err.Error(), "proxy handler is required") {
+		t.Fatalf("LaunchStandalone error = %v, want required proxy handler error", err)
+	}
+	if _, statErr := os.Stat(marker); !errors.Is(statErr, os.ErrNotExist) {
+		t.Fatalf("child started despite required proxy handler denial: stat error = %v", statErr)
+	}
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/sandbox/standalone_test.go:134]: This file carries a `//go:build linux` constraint (line 4), so these tests are only compiled on Linux and never reach the `launcher_darwin.go` / `launcher_other.go` stubs, so there are no non-Linux misleading failures to guard against. The three tests also exercise pre-namespace rejection logic (a nil required handler, `RequireNetNS` combined with `best_effort`, and the preflight requirements struct), all of which return before any namespace probe, so they pass on any Linux host without requiring `CLONE_NEWUSER`. Adding `skipIfStandaloneUnavailable` would wrongly skip real coverage on a runner that lacks namespace privileges. Not making this change.

File: internal/sandbox/bridge_test.go [hunk 2]
```diff
@@ -64,7 +65,7 @@ func TestBridgeProxy_ForwardsToUnixSocket(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	serveDone := make(chan struct{})
 	go func() {
-		bp.Serve(ctx)
+		_ = bp.Serve(ctx)
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/sandbox/bridge_test.go:68]: Deferring this one. The graceful nil-return path is now asserted directly in TestBridgeProxy_FailWakesServeWithRecordedError and TestBridgeProxy_FailAfterCloseIsDropped. The pre-existing forwarding tests synchronize deterministically through serveDone/cancel before reading their result, so adding a per-test nil assertion to each is low value against that coverage; tracking it as a minor test-quality follow-up rather than expanding this PR.

File: internal/sandbox/child_init.go [hunk 3]
```diff
@@ -206,7 +206,10 @@ func runInitWithBridge(command, env []string, workspace, socketPath string, sigC
 
 	ctx, cancel := context.WithCancel(context.Background())
 
-	go bridge.Serve(ctx)
+	// The MCP bridge does not yet supervise Serve's failure the way the
+	// standalone path does; explicitly discard it here to keep behavior
+	// unchanged. Hardening this path is tracked as a separate follow-up.
+	go func() { _ = bridge.Serve(ctx) }()
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/sandbox/child_init.go:212]: Deferring: this is the MCP proxy path (pipelock mcp proxy), a separate feature from the standalone guard lifecycle this PR hardens. Its bridge-death behavior is not regressed here, Serve was void before and this caller ignored it, and it now returns an error the MCP path still discards, preserving the prior behavior. Propagating fatal bridge failures into the MCP process lifecycle the way the standalone path now does is a real improvement but a heavier, separate change, tracked as a follow-up.

File: internal/sandbox/standalone_test.go [hunk 4]
```diff
@@ -179,6 +208,69 @@ func TestLaunchStandalone_BridgeProxyListens(t *testing.T) {
...
+	}
+	if got := info.Mode().Perm(); got != 0o700 {
+		t.Fatalf("control directory mode = %#o, want 0700", got)
+	}
+	// The name must be unpredictable, not the per-PID path an attacker could
+	// anticipate and pre-create or symlink in the sticky /tmp.
+	if predictable := fmt.Sprintf("/tmp/pipelock-sandbox-%d", os.Getpid()); dir == predictable {
+		t.Fatalf("control directory uses the predictable per-PID path: %s", dir)
+	}
+}
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply internal/sandbox/standalone_test.go:244]: Partially addressed: added a two-allocation check that a predictable implementation would fail. The negative rejection paths (symlink, foreign owner, unavailable metadata) cannot be exercised without permission to chown to another uid or an injectable FileInfo. Extracting the metadata validator so tests can inject invalid FileInfo values is a reasonable refactor but heavier than this PR, so the negative-path tests are deferred as a follow-up.

**Review by Celest-Blue — APPROVED** [review]
